File: fixtures/scale/php/medium/src/Repl/CommandRouter.php
<?php

namespace Calc\Repl;

require_once __DIR__ . '/HelpCommand.php';
require_once __DIR__ . '/VarsCommand.php';
require_once __DIR__ . '/HistoryCommand.php';
require_once __DIR__ . '/PrecisionCommand.php';
require_once __DIR__ . '/AngleCommand.php';
require_once __DIR__ . '/QuitCommand.php';

class CommandRouter
{
    public static function dispatch(string $line, ReplContext $context): string
    {
        $parts = explode(' ', ltrim($line, ':'));
        return match ($parts[0]) {
            'help' => HelpCommand::run($context, $parts),
            'vars' => VarsCommand::run($context, $parts),
            'history' => HistoryCommand::run($context, $parts),
            'precision' => PrecisionCommand::run($context, $parts),
            'angle' => AngleCommand::run($context, $parts),
            'quit' => QuitCommand::run($context, $parts),
            default => 'unknown command :' . $parts[0],
        };
    }
}

File: fixtures/scale/csharp/medium/Repl/CommandRouter.cs
namespace Calc.Repl
{
    public static class CommandRouter
    {
        public static string Dispatch(string line, ReplContext context)
        {
            var parts = line.TrimStart(':').Split(' ');
            switch (parts[0])
            {
                case "help":
                    return HelpCommand.Run(context, parts);
                case "vars":
                    return VarsCommand.Run(context, parts);
                case "history":
                    return HistoryCommand.Run(context, parts);
                case "precision":
                    return PrecisionCommand.Run(context, parts);
                case "angle":
                    return AngleCommand.Run(context, parts);
                case "quit":
                    return QuitCommand.Run(context, parts);
                default:
                    return "unknown command :" + parts[0];
            }
        }
    }
}
